File: bonus.jl
function pass()
    n = Iterators.countfrom(100)
    function isit(x)
        if (sum(digits(x))^3) == x
            return false
        else
            return true
        end
    end
    f = Iterators.dropwhile(isit, n)
    (t, a) = Iterators.peel(f)
    return t
end

"""
i7
@btime pass()
    14.831 μs (413 allocations: 45.17 KiB)
512
"""

File: bonus.py
import itertools
import timeit

def ps():
    n = itertools.count(100)
    def isit(x):
        if (((sum(int(d) for d in str(x))))**3 == x):
            return False
        else:
            return True
    return list(itertools.islice(itertools.dropwhile(isit, n), 1))

print(f"{(timeit.timeit(ps, number=1000)/1000)*(10**6)} microseconds")

"""
i7
290.39572598412633 microseconds
"""